mscore/masterpalette.cpp

```diff
@@ -263,5 +265,114 @@ void MasterPalette::changeEvent(QEvent *event)
             retranslate();
       }
 
+//---------------------------------------------------------
+//   keyPressEvent
+//---------------------------------------------------------
+
+void MasterPalette::keyPressEvent(QKeyEvent *event)
+      {
+      QTreeWidgetItem* currentItem = treeWidget->currentItem();
+      int idx = treeWidget->indexOfTopLevelItem(currentItem);
+      if (event->key() == Qt::Key_Down) {
+            if (idx >= 0 && idx < treeWidget->topLevelItemCount() - 1) {
+                  selectItem(treeWidget->topLevelItem(idx+1)->text(0));
+                  }
+            else if (currentItem == symbolItem) {
```

Comment:
I'm still not really comfortable with the special casing going on here.  Is it really not possible to have attributes to set and check, instead of special casing the individual palettes?

Reply:
I could subclass QTreeWidgetItem and store pointers to the next and previous item, which can be initialized in the addPalette function or in the masterPalette constructor. These pointers can be used in the keyPressEvent function to return the next/previous item.